numba/targets/arrayobj.py

```diff
@@ -1101,6 +1107,8 @@ def array_reshape(context, builder, sig, args):
     fail = builder.icmp_unsigned('==', ok, lc.Constant.int(ok.type, 0))
 
     with builder.if_then(fail):
+        # Release allocated array if the function error out
+        context.nrt_decref(builder, aryty, args[0])
```

Comment:
I now realize it is a bad fix for the problem that I didn't think through.  The actual problem is that cleanup code is not executed when an exception is raised.  As a result, the incref for the argument is not reversed.  I need to do a proper fix.